### PR TITLE
(PLUGIN-303) Added new property to MutliStageSelector widget

### PR DESCRIPTION
**Pull request by MEseifan**

### CODE REVIEW

File: cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx [hunk 1]
```diff
@@ -36,6 +37,7 @@ const styles = () => {
 
 interface IMultiStageSelectorWidgetProps {
   delimiter: string;
+  singleSelectOnly: boolean;
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx]: Should this be optional?

File: cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx [hunk 2]
```diff
@@ -58,28 +60,44 @@ const MultiStageSelectorBase: React.FC<IMultiStageSelectorProps> = ({
...
-  return (
-    <MultiSelect
-      value={value}
-      onChange={onChange}
-      widgetProps={multiSelectWidgetProps}
-      disabled={disabled}
-    />
-  );
+
+  if (singleSelectOnly || false) {
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx]: I don't think we need the "or" logic here.

File: cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx [hunk 3]
```diff
@@ -58,28 +60,44 @@ const MultiStageSelectorBase: React.FC<IMultiStageSelectorProps> = ({
...
   const inputSchema = objectQuery(extraConfig, 'inputSchema');
   const inputStages = getInputStages(inputSchema);
-  const { delimiter } = widgetProps;
-  const multiSelectWidgetProps = {
-    delimiter,
-    options: inputStages,
-    showSelectionCount: false,
-    emptyPlaceholder: 'Select input stage',
-  };
+  const { delimiter, singleSelectOnly } = widgetProps;
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx]: nit: if `singleSelectOnly` value is not provided set it to false here like this,

```
const { delimiter, singleSelectOnly = false } = widgetProps;
```

File: cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx [hunk 4]
```diff
@@ -58,28 +60,44 @@ const MultiStageSelectorBase: React.FC<IMultiStageSelectorProps> = ({
...
-      value={value}
-      onChange={onChange}
-      widgetProps={multiSelectWidgetProps}
-      disabled={disabled}
-    />
-  );
+
+  if (singleSelectOnly || false) {
+    const selectWidgetProps = {
+      options: inputStages.map((stage) => stage.label),
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx:75]: Can we reuse the same `getInputStages` function here?

**Review by ajainarayanan — APPROVED** [review]
Minor comments. Otherwise 👍

File: cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx (hunk 1 above)
Comment (MEseifan) [reply cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx]: I think so, but I was following the convention for `delimiter`. I think thats technically an optional parameter (at least it is for the underlying MultiSelect) but we didnt mark it as optional here. Wasnt sure if that was important somehow. I can make it optional.

File: cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx (hunk 2 above)
Comment (MEseifan) [reply cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx]: right...thanks!

File: cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx (hunk 3 above)
Comment (MEseifan) [reply cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx:63]: done

File: cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx (hunk 4 above)
Comment (MEseifan) [reply cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx:75]: not quite...Multselect expects options to be `{'id': '.', 'value': '.'}` meanwhile Select expects  `{'label': '.', 'value': '.'}` so I would need to move this logic into that function anyway to change the format

**Review by yukiej — APPROVED** [review]
lgtm!

File: cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx (hunk 4 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx:75]: Haa! That's a bad API. Ok can you file a jira for this. It's ok to merge now but that's bad.

**Comment by MEseifan** [comment]
@googlebot  rescan me!

**Pull request merged**